shared/src/main/java/com/couchbase/lite/Dictionary.java

```diff
@@ -356,7 +366,7 @@ public Date getDate(String key) {
      */
     @Override
     public Map<String, Object> toMap() {
-        Map<String, Object> result = map != null ? new HashMap<>(map) : new HashMap<String, Object>();
+        Map<String, Object> result = map != null ? new TreeMap<>(map) : new TreeMap<String, Object>();
```

Comment:
FYI, JSONObject is also used HashMap as a backing data.

https://android.googlesource.com/platform/dalvik/+/f2f7880a40aed1c8d9c27db49226e47396556aac/libcore/json/src/main/java/org/json/JSONObject.java